hq/app/aws/ec2/EC2.scala

```diff
@@ -56,7 +56,7 @@ object EC2 {
       sgResult <- TrustedAdvisorSGOpenPorts.getSGOpenPorts(supportClient)
       sgUsage <- getSgsUsage(sgResult, account)
       flaggedSgs = sgResult.flaggedResources.filter(_.status != "ok")
-    } yield flaggedSgs.map(sgOpenPortsDetail => sgOpenPortsDetail -> sgUsage.getOrElse(sgOpenPortsDetail.id, Set.empty))
+    } yield sortSecurityGroupsByInUse(flaggedSgs.map(sgOpenPortsDetail => sgOpenPortsDetail -> sgUsage.getOrElse(sgOpenPortsDetail.id, Set.empty)))
```

Comment:
This is for sure too complicated for a one-liner yield, but it's also possibly too much logic for an untested for-comp. Is there any reason the sort function doesn't take `flaggedSgs` so it can do the `map` (and `getOrElse`) step itself as well?

Reply:
yeah, can deal with that.